### PR TITLE
Refactor BoardInfoHelper and Improve Board Detection Mechanism

### PR DESCRIPTION
**Pull request by dariuszzbyrad**
This pull request addresses and resolves issues outlined in [GitHub Issue #421](https://github.com/Pi4J/pi4j-v2/issues/421), specifically the error encountered with the execution of commands via `sh -c` (`IOException: Cannot run program "sh": error=13, Permission denied`). It introduces a refactored approach to system command execution and board information detection while enhancing code maintainability and testability.

### **Changes Introduced:**

1.  **Removal of `sh -c` for Simple Commands:**

    -   Simplified command execution logic using `ProcessBuilder` directly for straightforward commands like `uptime`, avoiding unnecessary dependency on shell execution.
    -   For complex commands involving pipes (e.g., `cat /proc/cpuinfo | grep 'Revision' | awk '{print $3}'`), replaced shell commands with file readers and Java-based processing for better portability and reliability.
2.  **Improved Singleton Implementation in `BoardInfoHelper`:**

    -   Refactored the singleton to use the Bill Pugh Singleton Design for thread-safety and lazy initialization.
    -   Added a `reinitialize` method for controlled reinitialization, facilitating testing and debugging scenarios.
3.  **Encapsulation of Commands and System Properties:**

    -   Introduced centralized interfaces for system commands (`Command`) and properties (`SystemProperties`), ensuring commands and property keys are managed in one place with clear descriptions.
4.  **Enhanced Code Maintainability:**

    -   Improved JavaDocs for public methods and core components, enhancing clarity for contributors and maintainers.
    -   Modularized logic for better readability and testability (e.g., separate readers for CPU, memory, and board details).
5.  **Added Unit Tests:**

    -   Developed unit tests for board detection logic, specifically focusing on `getBoardModel` and related methods.
    -   Ensured file path overrides are testable, enabling reliable validation in different environments.
6.  **Improved Board Information Reading:**

    -   Replaced error-prone shell commands with efficient Java code for reading system files such as `/proc/cpuinfo` and `/proc/device-tree/model`.
    -   Incorporated `CommandResult` for consistent output across all data readers.

### CODE REVIEW

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public static CommandResult getCpuRevision() {
+        String outputMessage = StringUtil.EMPTY;
+        String errorMessage = StringUtil.EMPTY;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(cpuInfoFilePath))) {
+            String line;
+            // Read file line by line to locate the "Revision" entry.
+            while ((line = reader.readLine()) != null) {
+                if (line.startsWith("Revision")) {
```
Comment (eitch) [inline pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java]: Let's continue early and reduce the levels of nesting

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     *         - {@code errorMessage}: an error message if the entry is not found or if reading fails.
+     */
+    public static CommandResult getMemTotal() {
+        String errorMessage = StringUtil.EMPTY;
+        String memTotalLine = StringUtil.EMPTY;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(memInfoFilePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.startsWith("MemTotal:")) {
```
Comment (eitch) [inline pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java]: Let's continue early and reduce the levels of nesting

**Review by eitch — CHANGES_REQUESTED** [review]
Thanks for the exhaustive tests. This code looks good. I'd prefer less nesting. Otherwise all good

**Comment by FDelporte** [comment]
Indeed, great work! You're pushing this code up to a higher level! And I love the unit tests.
Let us know if the nesting can be reduced, and then we can quickly merge and have a 2.7.1-SNAPSHOT that can be tested.

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java (hunk 1 above)
Comment (dariuszzbyrad) [reply pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java]: I’ve simplified the loop by using an early continue to skip irrelevant lines, which reduces nesting - I hope this aligns with your suggestion.

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java (hunk 2 above)
Comment (dariuszzbyrad) [reply pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java]: Fixed, the same as in the CpuInfoReader class

**Review by FDelporte — APPROVED** [review]

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java (hunk 1 above)
Comment (eitch) [reply pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/CpuInfoReader.java]: perfect, thanks

File: pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java (hunk 2 above)
Comment (eitch) [reply pi4j-core/src/main/java/com/pi4j/boardinfo/datareader/MemInfoReader.java]: perfect, thanks

**Review by eitch — APPROVED** [review]

**Pull request merged**